### PR TITLE
Added TS type declarations from @types/react-native-share

### PR DESCRIPTION
**Pull request by taeh98**
This copies the type declaration files from @types/react-native-share into the repo and links them to package.json.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,13 +10,20 @@
...
     "ios",
     "windows",
     "RNShare.podspec"
   ],
+	"peerDependencies": {
+		"@types/react": "*",
+		"@types/react-native": "*",
+		"react": "*",
+		"react-native": "*"
+	},
```
Comment (MateusAndrade) [inline package.json:26]: Why do we need these declarations? This would work without it?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
+// Type definitions for react-native-share 3.3
+// Project: https://github.com/react-native-community/react-native-share#readme
```
Comment (MateusAndrade) [inline index.d.ts:2]: Maybe we could remove this and set a comment regarding the migration? What do you think?

**Review by MateusAndrade — CHANGES_REQUESTED** [review]

**Comment by MateusAndrade** [comment]
@taeh98 can you fill in a description on the PR?

File: package.json (hunk 1 above)
Comment (taeh98) [reply package.json]: I copied and pasted them from the commit in another React native project that added TS types. Now removed.

File: index.d.ts (hunk 2 above)
Comment (taeh98) [reply index.d.ts]: New comment now written.

**Comment by taeh98** [comment]
> @taeh98 can you fill in a description on the PR?

Done.

**Comment by MateusAndrade** [comment]
> > @taeh98 can you fill in a description on the PR?
> 
> Done.

Nice! Btw, we need to run a `yarn` to update the `yarn.lock`, otherwise, we will end with an error on the lock.

**Comment by taeh98** [comment]
> > > @taeh98 can you fill in a description on the PR?
> > 
> > 
> > Done.
> 
> Nice! Btw, we need to run a `yarn` to update the `yarn.lock`, otherwise, we will end with an error on the lock.

Done in https://github.com/react-native-share/react-native-share/pull/994/commits/3a96139b1a264427ca7e4fda5286f7bb814054ee

**Review by MateusAndrade — APPROVED** [review]
Thanks!!!

**Comment by DanielMarkiel** [comment]
Hi @MateusAndrade,
I just upgraded to v5.2.0 and I can't find type declarations in the package (no `index.d.ts` file in `node_modules/react-native-share`).
Are you sure they are exported correctly?

Ps. I tried to remove `node_modules` and install them again (using yarn v1.22.10) - same effect.

image

**Comment by MateusAndrade** [comment]
> Hi @MateusAndrade,
> I just upgraded to v5.2.0 and I can't find type declarations in the package (no `index.d.ts` file in `node_modules/react-native-share`).
> Are you sure they are exported correctly?
> 
> Ps. I tried to remove `node_modules` and install them again (using yarn v1.22.10) - same effect.
> 
> image

I think we need to publish then to npm with the other packages, we will need to change this: https://github.com/react-native-share/react-native-share/blob/f5521caa264455e2f777c4da20f5d158bcf6f0bd/package.json#L14-L20 adding the `*.ts` files

**Comment by mikehardy** [comment]
Also a types field I think? https://github.com/react-native-device-info/react-native-device-info/blob/7209edc41d8323796324f2ecc3459d50d4978308/package.json#L6

**Comment by MateusAndrade** [comment]
Fixing the missing publish config at #997

**Pull request merged**